pulsar-client/src/test/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImplTest.java

```diff
@@ -62,4 +71,43 @@ public void testGetStats() throws Exception {
         impl.getStats();
     }
 
+    @Test
+    public void multiTopicsInDifferentNameSpace() throws PulsarClientException {
+        List<String> topics = new ArrayList<>();
+        topics.add("persistent://public/default/MultiTopics1");
+        topics.add("persistent://public/test-multi/MultiTopics2");
+        topics.add("persistent://public/test-multi/MultiTopics3");
+        ClientConfigurationData conf = new ClientConfigurationData();
+        conf.setServiceUrl("pulsar://localhost:6650");
+        conf.setStatsIntervalSeconds(100);
+
+        PulsarClientImpl clientImpl = new PulsarClientImpl(conf);
+
+        Consumer consumer = clientImpl.newConsumer().topics(topics)
+                .subscriptionName("multiTopicSubscription")
+                .messageListener(new MessageListener<byte[]>() {
+                    @Override
+                    public void received(Consumer<byte[]> consumer, Message<byte[]> msg) {
+                       if("producer".equals(msg.getProducerName()) || "producer1".equals(msg.getProducerName())){
+                           String message = new String(msg.getData());
+                           assertTrue(message.contains("MultiTopics"));
```

Comment:
The assertion is done in another thread. The code might never run since you close consumer immediately after producers sent the records.

Please change the logic to use `#receive`. Add the logic to receive messages and verify them after line 109.

Reply:
I'v fixed ,PTAL.